pkg/controller/util/node/controller_utils.go

```diff
@@ -72,11 +71,6 @@ func DeletePods(ctx context.Context, kubeClient clientset.Interface, pods []*v1.
 			remaining = true
 			continue
 		}
-		// if the pod is managed by a daemonset, ignore it
-		if _, err := daemonStore.GetPodDaemonSets(pod); err == nil {
```

Comment:
Can you explain why this code is useless and how is it  related to https://github.com/kubernetes/kubernetes/pull/115840 ?

Reply:
Because this function is never used anywhere except in node_lifecycle_controller_test.go. Should I move it into this test file?

https://github.com/kubernetes/kubernetes/pull/115840 removed reference to `DeletePods` from node_lifecycle_controller.go.
The reason why `DeletePods` function is ignoring daemonset pods, I guess, is the old node life cycle controller behavior before taint/tolerance feature.

Reply:
Yes, please move it to the _test.go file as it's not used anywhere else.

Reply:
If `DeletePods ` not used, why can't it and its tests be simply removed?

Reply:
@bart0sh I find the root PR https://github.com/kubernetes/kubernetes/pull/36017 to add this test `TestPodStatusChange`. Since the old node lost eviction is replaced by taint based eviction, the test case is no longer needed anymore. So I add the second commit to delete all the code introduced by https://github.com/kubernetes/kubernetes/pull/36017 .